app/src/androidTest/java/com/sdp/movemeet/MainActivityTest.java

```diff
@@ -26,60 +55,89 @@
-        Intents.release();
-
+        onView(withId(R.id.nav_map)).perform(forceClick());
+        logout();
+    }*/
 
+    @Test
+    public void mainActivityToProfileActivity() {
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_edit_profile));
+        logout();
     }
 
     @Test
-    public void mainActivity_logout() {
-        Intents.init();
-
-        onView(withId(R.id.button_logout)).perform(click());
-
-        Intents.release();
+    public void mainActivityToActivityUpload() {
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_add_activity));
+        logout();
     }
 
     @Test
-    public void mainActivityToProfileActivity() {
-        Intents.init();
-
-        onView(withId(R.id.button_user_profile)).perform(click());
-
-        Intents.release();
+    public void mainActivityToStartActivity() {
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_start_activity));
+        logout();
     }
 
     @Test
-    public void mainActivityToFirebaseDebug() {
-        Intents.init();
-
-        onView(withId(R.id.firebaseButton)).perform(click());
-
-        Intents.release();
+    public void mainActivityGotoHome() {
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_start_activity));
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
+        // Open Drawer to click on navigation.
+        onView(withId(R.id.drawer_layout)).check(matches(isClosed(Gravity.LEFT))).perform(DrawerActions.open());
+        try{
+            Thread.sleep(500);
+        }catch(Exception e){}
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_home));
+        logout();
     }
 
     @Test
-    public void mainActivityToActivityUpload() {
-        Intents.init();
+    public void mainActivity_logout() {
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_logout));
+    }
 
-        onView(withId(R.id.goToActivityUploadButton)).perform(click());
+    public void logout() {
 
-        Intents.release();
-    }
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
 
-    @Test
-    public void mainActivityToStartActivity() {
-        Intents.init();
+        // Open Drawer to click on navigation.
+        onView(withId(R.id.drawer_layout))
+                .check(matches(isClosed(Gravity.LEFT))) // Left Drawer should be closed.
+                .perform(DrawerActions.open()); // Open Drawer
 
-        onView(withId(R.id.button2)).perform(click());
+        try{
+            Thread.sleep(500);
+        }catch(Exception e){}
 
-        Intents.release();
+        onView(withId(R.id.nav_view)).perform(NavigationViewActions.navigateTo(R.id.nav_logout));
```

Comment:
Ok so this button leads to the "logout()" method defined in MainActivity I guess 👍

Reply:
Yes, that's more or less it. Actually it does not lead directly to the activity hand logout. The function works more like this: whatever the page is, it opens the menu on the side and clicks on the "navigation" logout which currently uses a duplicated method in each class that does the logout but which, to be cleaner will be changed to use the method you defined in firebaseInteraction. This will remove the duplication of code and the logout method of FirebaseInteraction will be shared by all classes. I hope this is more or less clear